modules/basics/src/main/java/com/opengamma/basics/schedule/PeriodicSchedule.java

```diff
@@ -349,12 +348,22 @@ private void validate() {
    * @throws ScheduleException if the definition is invalid
    */
   public Schedule createSchedule() {
-    List<LocalDate> unadj = createUnadjustedDates();
+    List<LocalDate> unadj = generateUnadjustedDates();
     List<LocalDate> adj = applyBusinessDayAdjustment(unadj);
     RollConvention rollConv = getEffectiveRollConvention();
     List<SchedulePeriod> periods = new ArrayList<>();
-    for (int i = 0; i < unadj.size() - 1; i++) {
-      periods.add(SchedulePeriod.of(adj.get(i), adj.get(i + 1), unadj.get(i), unadj.get(i + 1)));
+    try {
+      // for performance, handle silly errors using exceptions
+      for (int i = 0; i < unadj.size() - 1; i++) {
+        periods.add(SchedulePeriod.of(adj.get(i), adj.get(i + 1), unadj.get(i), unadj.get(i + 1)));
+      }
+    } catch (IllegalArgumentException ex) {
+      // check dates to throw a better exception for duplicate dates in schedule
+      createUnadjustedDates();
+      createAdjustedDates();
+      // unknown exception
+      throw (ScheduleException) new ScheduleException(
```

Comment:
This is a very surprising piece of code. It's obvious what it does once you read the Javadocs for ``initCause``, but would it be clearer to do it in a slightly more conventional way:

    ScheduleException scheduleException = new ScheduleException(...);
    scheduleException.initCause(ex);
    throw scheduleException;